geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/geotools/SftArgResolver.scala

```diff
@@ -22,84 +23,100 @@ import scala.util.{Failure, Success, Try}
-        case Success(sft) => Some(sft)
-        case Failure(e) =>
-          logger.debug(s"Unable to parse sft spec from string $specArg with error ${e.getMessage}")
-          None
-      }
+  private[SftArgResolver] def parseSpecString(args: argTuple): resEither = {
+    Try(SimpleFeatureTypes.createType(
+        Option(args._2).getOrElse(throw new Throwable("Null feature name provided. " +
+          "Unable to parse spec string from null value.")),
+      args._1)
+    ) match {
+      case Success(sft) => Right(sft)
+      case Failure(e) => Left((s"Unable to parse sft spec from string ${args._1}.", e, SPECSTR))
     }
+  }
 
   // gets an sft based on a spec string
-  private[SftArgResolver] def parseSpecStringFile(specArg: String, name: String): Option[SimpleFeatureType] =
-    Option(specArg).map(new File(_)).flatMap { file =>
-      Try(SimpleFeatureTypes.createType (name, FileUtils.readFileToString(file))) match {
-        case Success(sft) => Some(sft)
-        case Failure(e) =>
-          logger.debug(s"Unable to parse sft spec from string $specArg with error ${e.getMessage}")
-          None
-      }
+  private[SftArgResolver] def parseSpecStringFile(args: argTuple): resEither =
+    Try(SimpleFeatureTypes.createType(
+      Option(args._2).getOrElse(throw new Throwable("Null feature name provided. " +
+        "Unable to parse spec string from null value.")),
+      FileUtils.readFileToString(new File(Option(args._1).getOrElse(""))))
+    ) match {
+      case Success(sft) => Right(sft)
+      case Failure(e) => Left((s"Unable to parse sft spec from string ${args._1} with error ${e.getMessage}", e, PATH))
     }
 
-  private[SftArgResolver] def parseConf(configStr: String, name: String): Option[SimpleFeatureType] =
+  private[SftArgResolver] def parseConf(configStr: String, name: String): Either[Throwable, SimpleFeatureType] = {
     Try {
       val sfts = SimpleSftParser.parseConf(ConfigFactory.parseString(configStr, parseOpts))
       if (sfts.size > 1) logger.warn(s"Found more than one SFT conf in arg '$configStr'")
       sfts.get(0)
     } match {
-      case Success(sft) if name == null || name == sft.getTypeName => Some(sft)
-      case Success(sft) => Some(SimpleFeatureTypes.renameSft(sft, name))
-      case Failure(e) => throw e
+      case Success(sft) if name == null || name == sft.getTypeName => Right(sft)
+      case Success(sft) => Right(SimpleFeatureTypes.renameSft(sft, name))
+      case Failure(e) => Left(e)
     }
+  }
 
   // gets an sft based on a spec conf string
-  private[SftArgResolver] def parseConfStr(specArg: String, name: String): Option[SimpleFeatureType] =
-    Try(parseConf(specArg, name)) match {
-      case Success(sftOpt) => sftOpt
-      case Failure(e) =>
-        logger.debug(s"Unable to parse sft spec from string $specArg as conf with error ${e.getMessage}")
-        None
+  private[SftArgResolver] def parseConfStr(args: argTuple): resEither =
```

Comment:
ResEither is actually, Either[(String, Throwable, Value), ReturnType] because we need to return the type that function is looking for so we can preserve that message if it matches our type guess. we could maybe just have Either[(Throwable, Value), ReturnType] and just pull the message from the exception but that doesn't really gain us anything.